backend/src/routes/api/messaging/get_groupchat.ts

```diff
@@ -0,0 +1,51 @@
+import { Request, Response } from "express";
+import GroupChats from "../../../models/Groupchat";
+import groupchat from "../../../types/groupchat";
+import Messaging from "../../../models/Messages";
+import {User} from "../../../models/User";
+import user from "../../../types/user";
+
+async function getGroupChatByID(req: Request, res: Response) {

```

Comment:
make sure to change this now, it requires VerifyToken authentication and then double check that the user is in the groupchat before giving the user all the messages of the groupchat